Description: Disable Clang >= 20 compiler options until packaged
The -fextend-variable-liveness=none flag was added to Chromium to work with newer versions of Clang (likely Clang 20+)
as -fextend-variable-liveness=none flag is not available in Clang<20 
It is a new compiler flag: https://releases.llvm.org/20.1.0/tools/clang/docs/ReleaseNotes.html#new-compiler-flags
This whole patch can be removed once we package Clang 20+ in the future.
Author: James Page <james.page@chainguard.dev>

--- a/build/config/compiler/BUILD.gn.orig	2025-05-12 16:11:29.949768459 +0100
+++ b/build/config/compiler/BUILD.gn	2025-05-12 16:12:14.385296607 +0100
@@ -313,7 +313,6 @@ config("compiler") {
     ":compiler_cpu_abi",
     ":compiler_codegen",
     ":compiler_deterministic",
-    ":clang_warning_suppression",
   ]
 
   # Here we enable -fno-delete-null-pointer-checks, which makes various nullptr
@@ -623,13 +622,6 @@ config("compiler") {
         default_toolchain != "//build/toolchain/cros:target") {
       cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
     }
-
-    # TODO(crbug.com/413427035): Remove once
-    # https://github.com/llvm/llvm-project/pull/136867/ is landed.
-    if (!is_win && !llvm_android_mainline &&
-        default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [ "-fextend-variable-liveness=none" ]
-    }
   }
 
   # C11/C++11 compiler flags setup.
